### PR TITLE
Add zio-http adapter without dependency on Tapir (a.k.a. `QuickAdapter`)

### PR DESCRIPTION
**Pull request by kyri-petrou**
As we've seen with recent benchmarks, using Tapir as an "intermediary" adds some overhead to response latencies. This PR adds the `QuickAdapter`, which uses `zio-http` and `jsoniter-scala` as the server & json implementations to provide the best possible performance out-of-the-box.

This adapter is still fully compliant with the "GraphQL over HTTP" spec, but doesn't support websockets. I included multipart / defer support, but I'm happy to remove that if we think that's too much for the "slim" variant. Let me know what you think!

Note that unlike the tapir adapters, this one doesn't have an equivalent `intercepted` method. This is mainly because creating zio-http middleware is fairly straightforward, and since that it would require changes to users code anyways (since the request type is different) I thought it's better to fully omit it

TODO:
- [x] Add example
- [x] Documentation

### CODE REVIEW

**Comment by frekw** [comment]
This is basically 95% similar to what we already use in prod! :D 👍

File: adapters/quick/src/main/scala/caliban/QuickAdapter.scala
Comment (kyri-petrou) [inline adapters/quick/src/main/scala/caliban/QuickAdapter.scala:1]: Hmmm I think this class might be getting a bit crowder, mixing handling of the requests with creating the HTTP interfaces. I think I might split all the request handling logic in a separate class, wdyt?

Comment (kyri-petrou) [reply adapters/quick/src/main/scala/caliban/QuickAdapter.scala:1]: Edit: I ended up making this change afterall as the code is much cleaner now

File: .circleci/config.yml
Comment (ghostdogpr) [inline .circleci/config.yml]: Hmm... the problem is that I usually update the docs with the new version *after* it's published (to avoid the docs being up with a version that doesn't exist yet)

File: adapters/quick/src/main/scala/caliban/GraphiQLAdapter.scala
Comment (ghostdogpr) [inline adapters/quick/src/main/scala/caliban/GraphiQLAdapter.scala]: Maybe weird to call it "adapter" since it's not really what it does with our definition of adapters. Maybe just `GraphiQLHandler`?

File: adapters/quick/src/main/scala/caliban/quick/package.scala
Comment (ghostdogpr) [inline adapters/quick/src/main/scala/caliban/quick/package.scala]: Can you add scaladocs? Not sure what `api` and `graphiql` are looking at types (are these paths? maybe it should be included in the names)

File: vuepress/docs/docs/adapters.md
Comment (ghostdogpr) [inline vuepress/docs/docs/adapters.md]: How about instead "If you want the best possible performance, make sure to check it out!"

File: vuepress/docs/docs/README.md
Comment (ghostdogpr) [inline vuepress/docs/docs/README.md]: Maybe add in the comment: "if you're just getting started, only import that one"

Comment (ghostdogpr) [inline vuepress/docs/docs/README.md]: Should we add an example of running the quick server right on the Getting Started page? Sounds like it would be good.

File: .circleci/config.yml
Comment (kyri-petrou) [reply .circleci/config.yml]: Would requiring the `release` step to be run first solve this?

I can add this to the task:

```yaml
requires:
  - release
```

Comment (ghostdogpr) [reply .circleci/config.yml]: That would work, but then what about when we want to update docs without a release?

Comment (kyri-petrou) [reply .circleci/config.yml]: Hmm one way would be to run the `npm run docs:build` locally and merge it into main, but that might be a bit too cumbersome.

File: vuepress/docs/docs/README.md
Comment (kyri-petrou) [reply vuepress/docs/docs/README.md]: I had to re-arrange a few things to make it coherent. Not sure if what I went with is the best approach - let me know

File: adapters/quick/src/main/scala/caliban/GraphiQLHandler.scala
Comment (ghostdogpr) [inline adapters/quick/src/main/scala/caliban/GraphiQLHandler.scala]: Maybe document that one since it's the one likely to be called

File: vuepress/docs/docs/README.md
Comment (ghostdogpr) [inline vuepress/docs/docs/README.md]:
```suggestion
First, add the following dependency to your `build.sbt` file:
```

File: vuepress/docs/docs/adapters.md
Comment (ghostdogpr) [inline vuepress/docs/docs/adapters.md]:
```suggestion
For example, we can serve our GraphQL api with minimal setup via a single command: 
```

File: vuepress/docs/docs/README.md
Comment (ghostdogpr) [reply vuepress/docs/docs/README.md]: Looks good!

File: .circleci/config.yml
Comment (ghostdogpr) [reply .circleci/config.yml]: I suggest for now, we put docs in a separate PR, and if we find a better solution for publishing docs we do it in a dedicated PR?

Comment (kyri-petrou) [reply .circleci/config.yml]: What about if we made the `microsite` workflow require manual approval? https://circleci.com/docs/workflows/#holding-a-workflow-for-a-manual-approval

We can "trial" it in a separate PR to see if we like it / works

Comment (ghostdogpr) [reply .circleci/config.yml]: I think the more fundamental issue with pushing "future" versions docs into the main branch is that until we release a new version, we can't update the docs anymore for other purposes. Any other change will have to wait a release. That's why I initially set it up this way, to have the freedom to update it anytime. In practice though, we usually update it more often when we change something and need to wait for the release. But the flexibility is nice 🤔

Comment (ghostdogpr) [reply .circleci/config.yml]: I guess an alternative would have to have docs versioning with an "unstable" version as well (like sttp/tapir), but it's work to set it up (not even sure vuepress can do that).

**Review by ghostdogpr — APPROVED** [review]

**Pull request merged**
